From 467f456c53a70320532a8854ba11236d829f9280 Mon Sep 17 00:00:00 2001
From: Jingtao Chen <jingtaoc@codeaurora.org>
Date: Thu, 28 May 2015 15:27:32 +0800
Subject: [PATCH 1/2] Bugfix for APBU-10448

For some specific mpegts file, first several DTS in pes header is
larger than PTS. Set DTS equals to PTS.

APBU-7629 is the same.

Change-Id: Icb3f970b45f62ef5c7fee20acecfc8cc4b408574
Signed-off-by: Jingtao Chen <jingtaoc@codeaurora.org>
---
 gst/mpegtsdemux/pesparse.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gst/mpegtsdemux/pesparse.c b/gst/mpegtsdemux/pesparse.c
index 1d7a3b6..c2328c5 100644
--- a/gst/mpegtsdemux/pesparse.c
+++ b/gst/mpegtsdemux/pesparse.c
@@ -166,6 +166,10 @@ mpegts_parse_pes_header (const guint8 * data, gsize length, PESHeader * res)
     READ_TS (data, res->DTS, bad_DTS_value);
     length -= 5;
 
+    /* Jingtao: pts should not be less than dts */
+    if (res->DTS > res->PTS)
+        res->DTS = res->PTS;
+
     GST_LOG ("DTS %" G_GUINT64_FORMAT " %" GST_TIME_FORMAT,
         res->DTS, GST_TIME_ARGS (MPEGTIME_TO_GSTTIME (res->DTS)));
   }
-- 
1.9.1

